libs/vkd3d/meta.c

```diff
@@ -620,6 +620,91 @@ void vkd3d_copy_image_ops_cleanup(struct vkd3d_copy_image_ops *meta_copy_image_o
+    VkRenderPassCreateInfo render_pass_info;
+    VkAttachmentDescription attachment_desc;
+    VkAttachmentReference attachment_ref;
+    VkSubpassDescription subpass_desc;
+    VkSubpassDependency subpass_dep;
+
+    render_pass_info.sType = VK_STRUCTURE_TYPE_RENDER_PASS_CREATE_INFO;
+    render_pass_info.pNext = NULL;
+    render_pass_info.flags = 0;
+    render_pass_info.attachmentCount = 1;
+    render_pass_info.pAttachments = &attachment_desc;
+    render_pass_info.subpassCount = 1;
+    render_pass_info.pSubpasses = &subpass_desc;
+    render_pass_info.dependencyCount = 1;
+    render_pass_info.pDependencies = &subpass_dep;
+
+    attachment_desc.loadOp = VK_ATTACHMENT_LOAD_OP_CLEAR;
+    attachment_desc.storeOp = VK_ATTACHMENT_STORE_OP_STORE;
+    attachment_desc.stencilLoadOp = VK_ATTACHMENT_LOAD_OP_DONT_CARE;
+    attachment_desc.stencilStoreOp = VK_ATTACHMENT_STORE_OP_DONT_CARE;
+    attachment_desc.format = format;
+    attachment_desc.samples = VK_SAMPLE_COUNT_1_BIT;
+    attachment_desc.initialLayout = VK_IMAGE_LAYOUT_UNDEFINED;
+    attachment_desc.finalLayout = VK_IMAGE_LAYOUT_PRESENT_SRC_KHR;
+    attachment_desc.flags = 0;
+
+    attachment_ref.attachment = 0;
+    attachment_ref.layout = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+
+    memset(&subpass_desc, 0, sizeof(subpass_desc));
+    subpass_desc.colorAttachmentCount = 1;
+    subpass_desc.pColorAttachments = &attachment_ref;
+    subpass_desc.pipelineBindPoint = VK_PIPELINE_BIND_POINT_GRAPHICS;
+
+    subpass_dep.srcSubpass = VK_SUBPASS_EXTERNAL;
+    subpass_dep.dstSubpass = 0;
+    subpass_dep.srcStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+    subpass_dep.dstStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
+    subpass_dep.srcAccessMask = 0;
+    subpass_dep.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
+    subpass_dep.dependencyFlags = 0;
+
+    return VK_CALL(vkCreateRenderPass(device->vk_device, &render_pass_info, NULL, render_pass));
+}
+
+static HRESULT vkd3d_meta_create_swapchain_pipeline(struct vkd3d_meta_ops *meta_ops,
+        const struct vkd3d_swapchain_pipeline_key *key, struct vkd3d_swapchain_pipeline *pipeline)
+{
+    const struct vkd3d_vk_device_procs *vk_procs = &meta_ops->device->vk_procs;
+    struct vkd3d_swapchain_ops *meta_swapchain_ops = &meta_ops->swapchain;
+    VkPipelineColorBlendAttachmentState blend_att;
+    VkPipelineColorBlendStateCreateInfo cb_state;
+    VkResult vr;
+
+    if ((vr = vkd3d_meta_create_swapchain_render_pass(meta_ops->device, key->format, &pipeline->vk_render_pass)))
+    {
+        ERR("Failed to create render pass, vr %d.\n", vr);
+        return hresult_from_vk_result(vr);
+    }
+
+    memset(&cb_state, 0, sizeof(cb_state));
+    memset(&blend_att, 0, sizeof(blend_att));
+    cb_state.sType = VK_STRUCTURE_TYPE_PIPELINE_COLOR_BLEND_STATE_CREATE_INFO;
+    cb_state.attachmentCount = 1;
+    cb_state.pAttachments = &blend_att;
+    blend_att.colorWriteMask = 0xf;
```

Comment:
`R_BIT | G_BIT | B_BIT | A_BIT`? I know it's annoying to type but I don't like magic numbers.

We could also make a `#define` for that somewhere since we run into this case quite a lot.

Reply:
Fixed.